wit/wasi-nn.wit

```diff
@@ -53,7 +67,15 @@ interface tensor {
         // Describe the type of element in the tensor (e.g., `f32`).
         ty: func() -> tensor-type;
 
-        // Return the tensor data.
+        // Describe where the tensor is currently located (e.g., `cpu`, `gpu`, `tpu`).
+        location: func() -> device;
+
+        // Move the tensor to a different device. This operation may result in an expensive data
+        // copy.
+        move-to: func(device: device) -> result<tensor, error>;
```

Comment:
@mtr's suggestion: rename to `copy-to` since that more accurately reflects the semantics here.

Reply:
@mingqiusun suggests `to`.